stac_fastapi/pgstac/stac_fastapi/pgstac/transactions.py

```diff
@@ -23,18 +24,38 @@ class TransactionsClient(AsyncBaseTransactionsClient):
     """Transactions extension specific CRUD operations."""
 
     async def create_item(
-        self, item: stac_types.Item, **kwargs
+        self, collection_id: str, item: stac_types.Item, **kwargs
     ) -> Optional[Union[stac_types.Item, Response]]:
         """Create item."""
+        body_collection_id = item.get("collection")
+        if body_collection_id is not None and collection_id != body_collection_id:
+            raise HTTPException(
+                status_code=409,
+                detail=f"Collection ID from path parameter ({collection_id}) does not match Collection ID from Item ({body_collection_id})",
+            )
+        item["collection"] = collection_id
         request = kwargs["request"]
         pool = request.app.state.writepool
         await dbfunc(pool, "create_item", item)
         return item
 
     async def update_item(
-        self, item: stac_types.Item, **kwargs
+        self, collection_id: str, item_id: str, item: stac_types.Item, **kwargs
     ) -> Optional[Union[stac_types.Item, Response]]:
         """Update item."""
+        body_collection_id = item.get("collection")
+        if body_collection_id is not None and collection_id != body_collection_id:
```

Comment:
`collection` is required field, I don't think `body_collection_id` would ever be `None`.

Reply:
The `collection` field is technically optional according to the [Item fields](https://github.com/radiantearth/stac-spec/blob/master/item-spec/item-spec.md#item-fields) spec:

> The id of the STAC Collection this Item references to (see [collection relation type](https://github.com/radiantearth/stac-spec/blob/master/item-spec/item-spec.md#relation-types)). This field is required if such a relation type is present and is not allowed otherwise.

Personally, I would prefer to be able to POST Items without a `collection` link/property since I know it's going to be autogenerated by the backend. If the `collection` property is required in the body then I need to also generate a Link with a `collection` rel type in order to validate the Item, which seems unnecessary since the Link and property will be overwritten anyway.